### PR TITLE
Code reliability improvements

### PR DESCRIPTION
**Pull request by ja6a**
Commits address each of the following issues. There are committed in the same order as the list. Refer to issue #18 

1. The method are not ordered in the most obvious way in the vm resource file - from the usability perspective.
2.   The logging level for spewing the object contents needs to be reduced to DEBUG level. In addition another simplified log entry for INFO level should be added.
3.   Remove global vm disk and vm network name structures from config. Replace with overloaded method.
4.   Remove unnecessary nil checks
5.   Remove test facilitation code for production code
6.   move all tests in resource_skytap_vm_hardware_test.go to resource_skytap_vm_test.go and delete
7.   rename adds and ids
8.   VM Create test is multistep that perform a modify, we need to divide this test in at least 4. test scenarios
    i) create vm
    ii) modify vm name
    iii) modify vm cpu
    iv) modify vm memory
9.   Tidy this line
10.   Improve this local variable name
12.   Check whether this validation is necessary
13. Retry if 422 received on create.

### CODE REVIEW

**Review by night0wl — APPROVED** [review]

File: skytap/config.go [hunk 1]
```diff
@@ -24,8 +24,6 @@ type SkytapClient struct {
 	vmsClient               skytap.VMsService
 	interfacesClient        skytap.InterfacesService
 	publishedServicesClient skytap.PublishedServicesService
-	vmDisks                 interface{}
-	vmNetworks              interface{}
```
Comment (pegerto) [inline skytap/config.go:28]: 👍

File: skytap/utils/response.go [hunk 2]
```diff
@@ -16,3 +16,14 @@ func ResponseErrorIsNotFound(responseError error) bool {
 
 	return false
 }
+
+// ResponseErrorIsUnprocessableEntity returns true if the error is UnprocessableEntity
+func ResponseErrorIsUnprocessableEntity(responseError error) bool {
```
Comment (pegerto) [inline skytap/utils/response.go]: Should not be this part of the SDK?

File: skytap/resource_skytap_vm.go [hunk 3]
```diff
@@ -521,15 +347,12 @@ func resourceSkytapVMRead(d *schema.ResourceData, meta interface{}) error {
...
-		}
+		d.Set("os_disk_size", *vm.Hardware.Disks[0].Size)
 	}
 
 	if len(vm.Hardware.Disks) > 1 {
 		var disks *schema.Set
-		if meta.(*SkytapClient).vmDisks == nil {
+		if vmDisks == nil {
 			// add the names
 			if v1 := d.Get("disk"); v1 != nil {
```
Comment (pegerto) [inline skytap/resource_skytap_vm.go]: not sure what is v1 in this context.

File: skytap/utils/response.go (hunk 2 above)
Comment (ja6a) [reply skytap/utils/response.go]: I presume you mean the whole retry on 422 routine - if so yes this should be moved to the API.

File: skytap/resource_skytap_vm.go (hunk 3 above)
Comment (ja6a) [reply skytap/resource_skytap_vm.go]: will change as follows: v1 -> diskResourcesMap. v2 -> diskResourceMap

File: skytap/utils/response.go (hunk 2 above)
Comment (ja6a) [reply skytap/utils/response.go]: done

File: skytap/resource_skytap_vm.go (hunk 3 above)
Comment (ja6a) [reply skytap/resource_skytap_vm.go]: renamed multiple nested variables to conform to the following format: 
1) a set e.g. diskSet
2) a list item, e.g. disk
3) a map, e.g. diskMap

**Review by pegerto — APPROVED** [review]

**Pull request merged**